arch-secure-boot

```diff
@@ -75,39 +90,48 @@ case "$1" in
             --add-section .initrd=ucode-initramfs.img --change-section-vma .initrd=0x3000000 \
             /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-unsigned.efi"
 
-        objcopy \
-            --add-section .osrel=/etc/os-release --change-section-vma .osrel=0x20000 \
-            --add-section .linux="/boot/vmlinuz-$KERNEL" --change-section-vma .linux=0x40000 \
-            --add-section .initrd="/boot/initramfs-$KERNEL-fallback.img" --change-section-vma .initrd=0x3000000 \
-            /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-recovery-unsigned.efi"
-
-        objcopy \
-            --add-section .osrel=/etc/os-release --change-section-vma .osrel=0x20000 \
-            --add-section .linux=/boot/vmlinuz-linux-lts --change-section-vma .linux=0x40000 \
-            --add-section .initrd=/boot/initramfs-linux-lts-fallback.img --change-section-vma .initrd=0x3000000 \
-            /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-recovery-lts-unsigned.efi"
+        for flavor in "${KERNEL#linux}" '-lts'; do
+            objcopy \
+                --add-section .osrel=/etc/os-release --change-section-vma .osrel=0x20000 \
+                --add-section .linux="/boot/vmlinuz-linux${flavor}" --change-section-vma .linux=0x40000 \
+                --add-section .initrd="/boot/initramfs-linux${flavor}-fallback.img" --change-section-vma .initrd=0x3000000 \
+                /usr/lib/systemd/boot/efi/linuxx64.efi.stub "${NAME%linux*}linux-recovery${flavor}-unsigned.efi"
+        done
 
-        for flavor in '' '-recovery' '-recovery-lts' '-efi-shell'; do
-            sbsign --key "$KEYSDIR/db.key" --cert "$KEYSDIR/db.crt" --output "${NAME}${flavor}.efi" "${NAME}${flavor}-unsigned.efi"
+        for flavor in "${KERNEL#linux}" "-recovery${KERNEL#linux}" '-recovery-lts' "${KERNEL#linux}-efi-shell"; do
+            sbsign --key "$KEYSDIR/db.key" --cert "$KEYSDIR/db.crt" --output "${NAME%linux*}linux${flavor}.efi" "${NAME%linux*}linux${flavor}-unsigned.efi"
         done
 
         sbsign --key "$KEYSDIR/db.key" --cert "$KEYSDIR/db.crt" /usr/lib/fwupd/efi/fwupdx64.efi
 
+        if [ -f "$ESP$SYSTEMD_BOOT" ]; then
+            sbsign --key "$KEYSDIR/db.key" --cert "$KEYSDIR/db.crt" --output "$ESP$SYSTEMD_SIGNED" "$ESP$SYSTEMD_BOOT"
+        fi
```

Comment:
For this to properly work, you would also need to add a hook so that this line is re-executed when systemd is updated, otherwise you will still be booting on the old version.